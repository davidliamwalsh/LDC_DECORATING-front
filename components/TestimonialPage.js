import { Component } from 'react'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import moment from 'moment'

class TestimonialPage extends Component {

  constructor (props) {
    super(props)

    this.TESTIMONIALS_QUERY = gql`
      query testimonials {
        testimonials  {
          slug
          id
          title
          body
          createdAt
        }
      }
    `
  }

  render () {
    const { testimonialSize } = this.props

    return <Query query={this.TESTIMONIALS_QUERY}>
      {({ loading, data }) => {
        if (loading) {
          return <>
            <p>Loading...</p>
          </>
        } else {
          return <>
            <div className='c-testimonials'>
              {data.testimonials.slice(0, testimonialSize).map((testimonial, index) => {
                return <div key={index}>
                  <h5>{testimonial.title}</h5>
                  <h5>{testimonial.body}</h5>
                  <h5>{moment(testimonial.createdAt).format('MMMM Do YYYY')}</h5>
                </div>
              })}
            </div>
          </>
        }
      }}
    </Query>
  }
}

export default TestimonialPage