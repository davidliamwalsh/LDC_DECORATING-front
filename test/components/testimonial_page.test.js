import TestimonialPage from '../../components/TestimonialPage'

describe('<TestimonialPage />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <TestimonialPage />
    )

    expect(component).toMatchSnapshot()
  })
})