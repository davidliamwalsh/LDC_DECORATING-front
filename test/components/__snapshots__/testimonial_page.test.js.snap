// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestimonialPage /> it renders correctly 1`] = `ShallowWrapper {}`;
